docs/concepts/feathr-concepts-for-beginners.md

```diff
@@ -0,0 +1,115 @@
+---
+layout: default
+title: Feathr Concepts for Beginners
+nav_order: 2
+---
+
+# Feathr Concepts for Beginners
+
+In this guide, we will cover the high level concepts for Feathr. Don't treat this as a user manual, instead treat this as blogpost to cover the highlevel motivations on why Feathr introduces those concepts.
+
+## What are `Observation` data, and why does Feathr need `key(s)`, `Anchor`, `Source`?
+
+In order to fully utilize Feathr's power, we need to understand the object models that Feathr is expecting.
+
+In Feathr, always think that there is some `Observation` dataset which is the central dataset that you will be using. The observation data set will usually have at least two columns: a timestamp column (indicating when this observation happens) and a column containing IDs, and with other possible fields
+
+For example, the `observation` dataset can be user click streams, credit card transactions, etc., while the data science team is trying to predict whether the user clicks on something, or whether the transcation is a fraud trasction, etc.
+
+Usually you will need addtional features to augment this `observation` dataset. For example, you want to augment the user click stream data by adding some historical features, such as the total amount that the user spent in the last one week. This additional dataset is usually in a different storage, say in your historical database, or data lake.
+
+In this case, how would we "link" the `observation` dataset, and the "additional dataset"? This is called [Feature Join](#why-does-feathr-need-feature-join-and-feature-query) in Feathr, but basically think this process as joining two tables.
+
+Since this is a table join process, we need to specify which `key(s)` that the join would happen. Those `keys` are usually some IDs, but can be others as well. In the above example, if we want to augment the user click stream data with user purchase history, we will use the user ID as `key`, so that the `click_stream` table and the `historical_buying` table can be joined together. Another case might be we want to get the historical blood pressure history for a patient, so `patient ID` will be used as key in that case. That's why you need to specify `keys` in Feathr `Feature`, becasue you will need to join your `Feature` with your `Observation Data` later on.
+
```

Comment:
let's use a example dataset. Users will get lost with so many words. Maybe a example is good here.